data/src/main/java/io/homeassistant/companion/android/data/integration/IntegrationRepositoryImpl.kt

```diff
@@ -52,6 +65,35 @@ class IntegrationRepositoryImpl @Inject constructor(
         throw IntegrationException()
     }
 
+    override suspend fun getRegistration(): DeviceRegistration {
+        return DeviceRegistration(
+            appVersion = localStorage.getString(PREF_APP_VERSION),
+            deviceName = localStorage.getString(PREF_DEVICE_NAME),
+            manufacturer = localStorage.getString(PREF_MANUFACTURER),
+            model = localStorage.getString(PREF_MODEL),
+            osVersion = localStorage.getString(PREF_OS_VERSION),
+            pushUrl = localStorage.getString(PREF_PUSH_URL),
+            pushToken = localStorage.getString(PREF_PUSH_TOKEN)
+        )
+    }
+
+    private suspend fun persistDeviceRegistration(deviceRegistration: DeviceRegistration) {
+        if (deviceRegistration.appVersion != null)
+            localStorage.putString(PREF_APP_VERSION, deviceRegistration.appVersion)
+        if (deviceRegistration.deviceName != null)
+            localStorage.putString(PREF_DEVICE_NAME, deviceRegistration.deviceName)
+        if (deviceRegistration.manufacturer != null)
+            localStorage.putString(PREF_MANUFACTURER, deviceRegistration.manufacturer)
```

Comment:
Why would we store these in local storage, can't we just fetch them from Android each time ?

Reply:
We could request it every time.  The reason I stored them is so that on subsequent updates we only need to pass what has changed rather than everything like the call requires.

Reply:
Migrated this to be injected.